### ISSUE TITLE
`__getattr__` needs to be ignored during operator availability checks

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pylance-release#1252]
## Environment data

-   Language Server version: 2021.5.1
-   OS and version: Windows 7
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.8 Anaconda

## Sample

```python
class C:
    def __getattr__(self, name: str, /):
        if name == "__add__":
            return lambda _: 0

a = C()
a.__add__  # Implied type: (_: Unknown) -> Literal[0] | None

b = a + 0  # Implied type: Literal[0]
```

## Expected behavior

The last line should be forbidden.

The internal logic of built-in operators / wrapper methods (for example: len(...)) calls the method defined on the class but not the instance, so `__getattr__` will not work.

If you run the sample, you will probably get

> TypeError: unsupported operand type(s) for +: 'C' and 'int'

on the last line.

## Actual behavior

No errors reported.

**Comment by erictraut** [comment]
Moving this issue to pyright, since it's a core type checker issue.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.5.3, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202153-19-may-2021
